extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergTableLocationProviderBase.java

```diff
@@ -0,0 +1,97 @@
+    final IcebergCatalogAdapter adapter;
+    final TableIdentifier tableIdentifier;
+
+    private TableDataRefreshService.CancellableSubscriptionToken subscriptionToken;
+
+    public IcebergTableLocationProviderBase(
+            @NotNull final TK tableKey,
+            @NotNull final IcebergBaseLayout locationKeyFinder,
+            @NotNull final TableLocationFactory<TK, TLK> locationFactory,
+            @Nullable final TableDataRefreshService refreshService,
+            @NotNull final IcebergCatalogAdapter adapter,
+            @NotNull final TableIdentifier tableIdentifier) {
+        super(tableKey, refreshService != null);
+        this.locationKeyFinder = locationKeyFinder;
+        this.locationFactory = locationFactory;
+        this.refreshService = refreshService;
+        this.adapter = adapter;
+        this.tableIdentifier = tableIdentifier;
+    }
+
+    // ------------------------------------------------------------------------------------------------------------------
+    // AbstractTableLocationProvider implementation
+    // ------------------------------------------------------------------------------------------------------------------
+
+    /**
+     * Update the table location provider with the latest snapshot from the catalog.
+     */
+    public abstract void update();
+
+    /**
+     * Update the table location provider with a specific snapshot from the catalog. If the {@code snapshotId} is not
+     * found in the list of snapshots for the table, an {@link IllegalArgumentException} is thrown. The input snapshot
+     * must also be newer (higher in sequence number) than the current snapshot or an {@link IllegalArgumentException}
+     * is thrown.
+     *
+     * @param snapshotId The identifier of the snapshot to use when updating the table.
+     */
+    public abstract void update(final long snapshotId);
+
+    /**
+     * Update the table location provider with a specific snapshot from the catalog. The input snapshot must be newer
+     * (higher in sequence number) than the current snapshot or an {@link IllegalArgumentException} is thrown.
+     * 
+     * @param snapshot The snapshot to use when updating the table.
+     */
+    public abstract void update(final Snapshot snapshot);
+
+    @Override
+    @NotNull
+    protected TableLocation makeTableLocation(@NotNull final TableLocationKey locationKey) {
+        // noinspection unchecked
+        return locationFactory.makeLocation((TK) getKey(), (TLK) locationKey, refreshService);
+    }
+
+    // ------------------------------------------------------------------------------------------------------------------
+    // SubscriptionAggregator implementation
+    // ------------------------------------------------------------------------------------------------------------------
+
+    @Override
+    protected final void activateUnderlyingDataSource() {
+        subscriptionToken = refreshService.scheduleTableLocationProviderRefresh(this);
+    }
```

Comment:
This only makes sense in the automatically-refreshing version.
Manual-refreshing, `supportsSubscriptions`, but `activateUnderlyingDataSource` should be a no-op.